lib/drivers/weitai/weiTaiUSB.js

```diff
@@ -0,0 +1,1565 @@
+/*
+ * == BSD2 LICENSE ==
+ * Copyright (c) 2019, Tidepool Project
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the associated License, which is identical to the BSD 2-Clause
+ * License as published by the Open Source Initiative at opensource.org.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the License for more details.
+ *
+ * You should have received a copy of the License along with this program; if
+ * not, you can obtain one from Tidepool Project at tidepool.org.
+ * == BSD2 LICENSE ==
+ */
+
+import _ from 'lodash';
+import sundial from 'sundial';
+
+import crypto from 'crypto';
+import TZOUtil from '../../TimezoneOffsetUtil';
+import UsbDevice from '../../usbDevice';
+import {
+  cRC8,
+  packFrame,
+  unpackFrame,
+  uintFromArrayBuffer,
+  formatString,
+} from './utils';
+
+const { promisify } = require('util');
+
+const isBrowser = typeof window !== 'undefined';
+// eslint-disable-next-line no-console
+const debug = isBrowser ? require('bows')('WeitaiUSBDriver') : console.log;
+
+const usb = require('usb');
+
+class WeitaiUSB {
+  constructor(cfg) {
+    this.cfg = cfg;
+  }
+
+  static get TIMEOUT() {
+    return 5000;
+  }
+
+  async openDevice(deviceInfo, cb) {
+    this.usbDevice = new UsbDevice(deviceInfo);
+    try {
+      this.usbDevice.device.open(false); // don't auto-configure
+    } catch (err) {
+      debug(err);
+      return cb({ code: 'E_SERIAL_CONNECTION' }, null);
+    }
+    this.usbDevice.device.reset(() => {
```

Comment:
This `reset()` command is what prevents the driver from working on Linux correctly. If you remove this, you can enable Linux support again.